This patch changes the test script ifconfig.sh to use `ipadm` instead
of `ifconfig` on Solaris 11.1 and above to provide a cleaner and more
accurate and specifically named interface configuration.

The patch will be provided upstream.

--- bind-9.16.29/bin/tests/system/ifconfig.sh
+++ bind-9.16.29/bin/tests/system/ifconfig.sh
@@ -70,7 +70,7 @@ *-sun-solaris2.[6-7])
 				ifconfig lo0:$int 10.53.$i.$ns \
 					netmask 0xffffffff up
 				;;
-			    *-*-solaris2.[8-9]|*-*-solaris2.1[0-9])
+			    *-*-solaris2.[8-9]|*-*-solaris2.10)
 				/sbin/ifconfig lo0:$int plumb
 				/sbin/ifconfig lo0:$int 10.53.$i.$ns up
 				/sbin/ifconfig lo0:$int mtu 1500
@@ -78,6 +78,17 @@ /sbin/ifconfig lo0:$int inet6 plumb
 				[ "$ipv6" ] && /sbin/ifconfig lo0:$int \
 					inet6 fd92:7065:b8e:${ipv6}ff::$ns up
 				;;
+			    *-*-solaris2.1[0-9])
+				/sbin/ipadm create-addr -t -T static \
+					    -a 10.53.$i.$ns lo0/v4$i$ns ||
+				    echo failed lo0/v4$i$ns
+				[ "$ipv6" ] && \
+				    /sbin/ipadm create-addr -t \
+						-T static \
+						-a fd92:7065:b8e:${ipv6}ff::$ns \
+						lo0/v6$i$ns ||
+					echo failed lo0/v6$i$ns
+				;;
 			    *-*-linux*)
                                 if [ $use_ip ]; then
                                         ip address add 10.53.$i.$ns/24 \
@@ -173,12 +184,18 @@ ;;
 			    *-sun-solaris2.[6-7])
 				ifconfig lo0:$int 10.53.$i.$ns down
 				;;
-			    *-*-solaris2.[8-9]|*-*-solaris2.1[0-9])
+			    *-*-solaris2.[8-9]|*-*-solaris2.10)
 				ifconfig lo0:$int 10.53.$i.$ns down
 				ifconfig lo0:$int 10.53.$i.$ns unplumb
 				ifconfig lo0:$int inet6 down
 				ifconfig lo0:$int inet6 unplumb
 				;;
+			    *-*-solaris2.1[1-9])
+				ipadm delete-addr lo0/v4$i$ns ||
+				    echo failed lo0/v4$i$ns
+				[ "$ipv6" ] && ipadm delete-addr lo0/v6$i$ns ||
+					echo failed lo0/v6$i$ns
+				;;
 			    *-*-linux*)
                                 if [ $use_ip ]; then
                                         ip address del 10.53.$i.$ns/24 \
